### PR TITLE
Research Agda

### PR DESCRIPTION
**Pull request by fl3pp**

### CODE REVIEW

File: SA/design_concept/content/tool_research/agda.typ [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+correctness of the code, but also to check incomplete programs. Additional
+context, such as checker errors or incomplete expressions, is then provided
+in a separate window inside the editor. 
+
+Using this tooling it is, for example, possible to insert a hole in the
+code, called a _metavariable_, which the checker will detect and display
+alongside the expected type in the context-window, as shown below.
+
+#figure(
+  image("../../static/agda-context-window.png", width: 70%),
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:19]: I think the actual image is currently missing.

File: SA/design_concept/content/tool_research/agda.typ [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  caption: [Screenshot of Agda context-window after inserting a hole in an expression using the `?` sign]
+)
+
+The agda language server is also able to provide automatic case-splitting
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:23]:
```suggestion
The Agda language server is also able to provide automatic case-splitting
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+for pattern matching. During this process, the checker will determine which
+cases are even possible, and present only those to the user. Thus, the user
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:25]:
```suggestion
cases are possible, and present only those to the user. Thus, the user
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+is not only freed from having to remember and type out all cases, but also
+from having to determine which cases are even possible.
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:27]:
```suggestion
from having to determine which cases are possible at all.
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 5]
```diff
@@ -0,0 +1,33 @@
+= Agda
+
+Even though Agda
+#footnote("https://wiki.portal.chalmers.se/agda/pmwiki.php") isn't a visual
+programming environment, the tools for it have been designed in an
+interactive way and are of interest for this project.
+
+The basis for these tools is a language-server, which in combination with
+the powerful checker and editor extensions, allows to not only verify the
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:9]:
```suggestion
the powerful checker and editor extensions, not only allows to verify the
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+correctness of the code, but also to check incomplete programs. Additional
+context, such as checker errors or incomplete expressions, is then provided
+in a separate window inside the editor. 
+
+Using this tooling it is, for example, possible to insert a hole in the
+code, called a _metavariable_, which the checker will detect and display
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:15]:
```suggestion
code, a so-called _metavariable_, which the checker will detect and display
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+alongside the expected type in the context-window, as shown below.
+
+#figure(
+  image("../../static/agda-context-window.png", width: 70%),
+  caption: [Screenshot of Agda context-window after inserting a hole in an expression using the `?` sign]
+)
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:21]:
```suggestion
) <agda_context_window>
```

File: SA/design_concept/content/tool_research/agda.typ [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+The basis for these tools is a language-server, which in combination with
+the powerful checker and editor extensions, allows to not only verify the
+correctness of the code, but also to check incomplete programs. Additional
+context, such as checker errors or incomplete expressions, is then provided
+in a separate window inside the editor. 
+
+Using this tooling it is, for example, possible to insert a hole in the
+code, called a _metavariable_, which the checker will detect and display
+alongside the expected type in the context-window, as shown below.
```
Comment (lstreckeisen) [inline SA/design_concept/content/tool_research/agda.typ:16]:
```suggestion
alongside the expected type in the context-window, as shown in @agda_context_window.
```

**Review by lstreckeisen — APPROVED** [review]

File: SA/design_concept/content/tool_research/agda.typ (hunk 1 above)
Comment (fl3pp) [reply SA/design_concept/content/tool_research/agda.typ:19]: added

**Pull request merged**